lib/info.c

```diff
@@ -588,7 +588,7 @@ int vorbis_analysis_headerout(vorbis_dsp_state *v,
   oggpack_buffer opb;
   private_state *b=v->backend_state;
 
-  if(!b||vi->channels<=0){
+  if(!b||vi->channels<=0||vi->channels>256){
```

Comment:
This should be `> 255` I think?

Do you have a way for vi->channels to fail this check? The value is initialized from an one-byte read in the info packet data, so memory corruption is the only way I see for a value >255 to occur.